File: examples/AODV/updateExpiryTimer.m
function updateExpiryTimer(tableID)

global routing_table
global verbose

dataTimer = ttGetData('TimerTask');

if (dataTimer.expTimer == 1)
  ttRemoveTimer('exptimer');
end

mintime = inf;
table = routing_table{tableID};

for k = 1:length(table)
  if ((table{k}.exptime < mintime) & strcmp(table{k}.state,'valid'))
    mintime = table{k}.exptime;
  end
end

if (mintime < inf)
  ttCreateTimer('exptimer', mintime, 'timer_handler');
  dataTimer.expTimer = 1;
  
  noww = ttCurrentTime;
  
  if verbose
    disp(['Time: ' num2str(noww) ' Node#' num2str(tableID) ' updating expiry timer to time: ' num2str(mintime)]);
  end
    
else
  dataTimer.expTimer = 0;
end
    
ttSetData('TimerTask', dataTimer); 

File: examples/RUNES_demo/node/updateExpiryTimer.cpp
#define INF 1.0E+15 

#define EPS 1.0E-15



void updateExpiryTimer(RouteEntry *table, Exp_Data *timerdata, int nodeID) {

  

  double mintime = INF; // defined in ttkernel.h

  double now;

  int i;

  

  if (timerdata->expTimer) {

    ttRemoveTimer("exptimer");

  }

  

  for (i=0; i<NBR_AODV; i++) {

    if (table[i].valid) {

      if (table[i].exptime < mintime) {

	mintime = table[i].exptime;

      }

    }

  }

  

  if (mintime < INF-100.0) {

    ttCreateTimer("exptimer", mintime, "timer_handler");

    timerdata->expTimer = 1;

    

    now = ttCurrentTime();

    

    if (VERBOSE) {

      mexPrintf("Time: %f Node#%d updating expiry timer to time: %f\n", now, nodeID, mintime);

    }

    

  } else {

    timerdata->expTimer = 0;

  }



}

